Swagger Integration

nguyenfilip opened the pull request:
This is merge of Swagger development branch into master.

barnabycourt commented:
@nguyenfilip Please squash the commits for this PR

barnabycourt commented:
@nguyenfilip Looking at the entitlements resource in the swagger-ui. There is one endpoint at /entitlements/{dbid} and another at /entitlements/{entitlement_id}. I think both of those should be {entitlement_id}. What is the correct method to fix those? If it should be done in follow-on BZ's/PRs that's fine.

nguyenfilip commented:
@barnabycourt The non consistency of some resource has been in the Resources even before. Will fix this along the way with this PR to {entitlement_id}.

To make the squash easier I created a different branch and squashed there. I squashed it into 4 commits. Will close this PR and the replacement PR is #1245
